packages/react/src/errorboundary.tsx

```diff
@@ -66,6 +66,25 @@ const INITIAL_STATE = {
   eventId: null,
 };
 
+function setCause(error: Error & { cause?: Error }, cause: Error): void {
```

Comment:
This looks weird to me:
- fn signature: `setCause(error, cause)`
- usage: `setCause(error, errorBoundaryError);`


`errorBoundaryError` is NOT the cause of the `error` being captured by the boundary. `errorBoundaryError` is an artificial error that you created at the top of the tree.

I have no idea why you try to put this error as a cause deeply in the error stack, but that looks really suspicious to me. errorBoundaryError should rather be the parent top-level error, not the most deeply nested cause.